### PR TITLE
chore: [SDP-6090] add support for OIDC authentication

### PR DESCRIPTION
**Pull request by stephenc**
Also:

* switch to a new release model
* ensure that the infrastructure is completely parameterised

### CODE REVIEW

File: .cloudbees/staging/action.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+apiVersion: automation.cloudbees.io/v1alpha1
```
Comment (alexeysas) [inline .cloudbees/staging/action.yml:1]: This repo is public - right?   Not sure if it good idea to have environment specific  internal folder "staging" there?

File: .cloudbees/workflows/workflow.yml [hunk 2]
```diff
@@ -9,19 +9,41 @@ on:
...
+        uses: cloudbees-io/checkout@v1
+      - name: Self Test
+        uses: docker://golang:1.21.7
+        run: |
+          make format sync check-git-status test
+      - id: dockerconfig
+        name: Configure container registry credentials
+        uses: cloudbees-io/configure-oci-credentials@v0
+        with:
+          registry: ${{ vars.STAGING_DOCKER_REGISTRY }}
```
Comment (alexeysas) [inline .cloudbees/workflows/workflow.yml:24]: Not sure if it good idea to pin public repo actions to staging specific variables.

File: .cloudbees/staging/action.yml (hunk 1 above)
Comment (stephenc) [reply .cloudbees/staging/action.yml:1]: it's the only way to self-test without requiring a follow-up change that cannot be tested

File: .cloudbees/workflows/workflow.yml (hunk 2 above)
Comment (stephenc) [reply .cloudbees/workflows/workflow.yml:24]: A different environment can use a different value. This way I am not hard coding

File: .cloudbees/staging/action.yml (hunk 1 above)
Comment (alexeysas) [reply .cloudbees/staging/action.yml:1]: Ok fine with me.

**Review by alexeysas — APPROVED** [review]

File: .cloudbees/workflows/workflow.yml (hunk 2 above)
Comment (alexeysas) [reply .cloudbees/workflows/workflow.yml:24]: Ok we an proceed - at least do not see better solution for now.

**Pull request merged**